src/main/scala/org/allenai/pipeline/S3Artifact.scala

```diff
@@ -1,23 +1,26 @@
 package org.allenai.pipeline
 
 import org.allenai.common.Logging
-import org.allenai.pipeline
 
 import com.amazonaws.AmazonServiceException
 import com.amazonaws.auth.BasicAWSCredentials
 import com.amazonaws.services.s3.AmazonS3Client
-import com.amazonaws.services.s3.model.{ CannedAccessControlList, PutObjectRequest, ObjectMetadata }
+import com.amazonaws.services.s3.model.{ CannedAccessControlList, ObjectMetadata, PutObjectRequest }
 
-import java.io.{ InputStream, File, FileOutputStream }
+import java.io.{ File, FileOutputStream, InputStream }
 import java.net.URI
 
-case class S3Config(service: AmazonS3Client, bucket: String)
+case class S3Config(initService: () => AmazonS3Client, bucket: String) {
+  @transient lazy val service = initService()
+}
 
 object S3Config {
-  def apply(accessKey: String, secretAccessKey: String, bucket: String): S3Config =
-    S3Config(new AmazonS3Client(new BasicAWSCredentials(accessKey, secretAccessKey)), bucket)
-
-  def apply(bucket: String): S3Config = S3Config(new AmazonS3Client(), bucket)
+  def apply(accessKey: String, secretAccessKey: String, bucket: String): S3Config = {
+    S3Config(() => new AmazonS3Client(new BasicAWSCredentials(accessKey, secretAccessKey)), bucket)
+  }
+  def apply(bucket: String): S3Config = {
+    S3Config(() => new AmazonS3Client(), bucket)
```

Comment:
Plan is to set up roles so that the no-arg `new AmazonS3Client()` constructor will work on the slave nodes.  The we can pass around `S3Config` instances that use the no-arg constructor.